internal/core/resolved.go

```diff
@@ -102,7 +102,8 @@ type GatewayID struct {
 }
 
 type ResolvedGateway struct {
-	ID        GatewayID
-	Meta      map[string]string
-	Listeners []ResolvedListener
+	ID             GatewayID
+	Meta           map[string]string
+	Listeners      []ResolvedListener
+	MaxConnections *uint32
```

Comment:
The `ResolvedGateway` is what we have access to when deriving the `ingress-gateway` config entry that we're sending to Consul, so it needs to be aware of any config that we depend on there. This is internal, so it can evolve in the future if we wind up having other connection management config values to pipe through.